Gems/Multiplayer/Code/Source/Components/NetworkDebugPlayerIdComponent.cpp

```diff
@@ -0,0 +1,174 @@
+
+        AZStd::string playerIdText = AZStd::string::format("Player %i", GetPlayerId());
+        m_fontDrawInterface->DrawScreenAlignedText2d(m_drawParams, playerIdText);
+
+        if (IsNetEntityRoleAutonomous())
+        {
+            const auto controller = static_cast<NetworkDebugPlayerIdComponentController*>(GetController());
+            if (!controller)
+            {
+                AZ_Assert(false, "NetworkDebugPlayerIdComponent failed to access its multiplayer controller.");
+                return;
+            }
+
+            const float textHeight = m_fontDrawInterface->GetTextSize(m_drawParams, playerIdText).GetY();
+            const float lineSpacing = 0.5f * textHeight;
+            m_drawParams.m_position.SetY(m_drawParams.m_position.GetY() + textHeight + lineSpacing);
+
+            AZStd::string playerCountText = AZStd::string::format("Player Count: %i", controller->GetConnectionCount());
+            m_fontDrawInterface->DrawScreenAlignedText2d(m_drawParams, playerCountText);
+        }
+    }
+
+    NetworkDebugPlayerIdComponentController::NetworkDebugPlayerIdComponentController(NetworkDebugPlayerIdComponent& parent)
+        : NetworkDebugPlayerIdComponentControllerBase(parent)
+    {
+    }
+
+    void NetworkDebugPlayerIdComponentController::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        # if AZ_TRAIT_SERVER
+            m_networkInterface = AZ::Interface<AzNetworking::INetworking>::Get()->RetrieveNetworkInterface(AZ::Name(MpNetworkInterfaceName));
+            if (IsNetEntityRoleAuthority())
+            {
+                uint32_t currentConnectionCount = m_networkInterface->GetConnectionSet().GetConnectionCount();
+                //SetConnectionCount(currentConnectionCount);
+
+                // Multiplayer system doesn't directly track the number of players.
+                // Instead just assign this player an id by checking how many machines are already connected to this host.
+                // Note 1: This doesn't support reassigning player-ids to rejoining players
+                // Note 2: This doesn't consider multi-server connections; it's possible that connection count includes other servers (not just player machines)
+                // Note 3: Client-server player count will be wrong by -1; client-server has its own player without any connections.
+                SetPlayerId(currentConnectionCount); 
+
+                AZ::TickBus::Handler::BusConnect();
+            }
+        #endif
+    }
+
+    void NetworkDebugPlayerIdComponentController::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        if (IsNetEntityRoleAuthority())
+        {
+            AZ::TickBus::Handler::BusDisconnect();
+        }
+    }
+
+    void NetworkDebugPlayerIdComponentController::OnTick([[maybe_unused]]float deltaTime, [[maybe_unused]]AZ::ScriptTimePoint time)
```

Comment:
Updating this every tick seems a bit brutal.. do we have an event for connect/disconnect on the server?

Reply:
Unfortunately, there's no event for connect/disconnect. If I added functionality to Multiplayer System, I would bypass connection count altogether and actually track OnPlayerJoin and OnPlayerLeave events.

Reply:
Oops, nope, you're right... there is a connection/disconnect event. So the on startup I'll grab the connection count directly, and then begin listening for new connections. Stay tuned!

Reply:
And don't bother trying to increment/decrement based on those events.. every time the event pops just grab the source of truth connection count from the network interface.

Reply:
fixed! thank you